packages/web-app-files/src/components/SideBar/SideBar.vue

```diff
@@ -200,94 +199,61 @@ export default defineComponent({
-    },
-    highlightedFileFavorite() {
-      return this.highlightedFile?.starred
-    },
     highlightedFileIsSpace() {
       return this.highlightedFile?.type === 'space'
     }
   },
   watch: {
-    highlightedFile(newFile, oldFile) {
-      if (!this.isSingleResource) {
-        return
-      }
-
-      if (oldFile && isEqual(newFile, oldFile)) {
-        return
-      }
-
-      // FIXME: isSpacesProjectsLocation resolves "true" within a project?!
-      const isSpacesProjectsLocation = this.isSpacesProjectsLocation && !this.currentStorageId
-      const loadShares =
-        !!oldFile ||
-        isSpacesProjectsLocation ||
-        this.isSharedWithMeLocation ||
-        this.isSharedWithOthersLocation ||
-        this.isSharedViaLinkLocation ||
-        this.isSearchLocation ||
-        this.isFavoritesLocation
-      this.fetchFileInfo(loadShares)
-    },
-
-    highlightedFileThumbnail(thumbnail) {
-      this.$set(this.selectedFile, 'thumbnail', thumbnail)
-    },
+    highlightedFile: {
+      handler(newFile, oldFile) {
+        const noChanges = oldFile && isEqual(newFile, oldFile)
+        if (!this.isSingleResource || !this.highlightedFile || noChanges) {
+          return
+        }
 
-    highlightedFileFavorite(starred) {
-      this.$set(this.selectedFile, 'starred', starred)
-    }
-  },
-  async created() {
-    if (!this.areMultipleSelected) {
-      await this.fetchFileInfo(false)
+        // FIXME: isSpacesProjectsLocation resolves "true" within a project?!
+        const isSpacesProjectsLocation = this.isSpacesProjectsLocation && !this.currentStorageId
+        const loadShares =
+          !!oldFile ||
+          isSpacesProjectsLocation ||
+          this.isSharedWithMeLocation ||
+          this.isSharedWithOthersLocation ||
+          this.isSharedViaLinkLocation ||
+          this.isSearchLocation ||
+          this.isFavoritesLocation
+        this.fetchFileInfo(loadShares)
+      },
+      deep: true
     }
   },
   methods: {
     ...mapActions('Files', ['loadSharesTree']),
 
     async fetchFileInfo(loadShares) {
-      if (!this.highlightedFile) {
-        this.selectedFile = {}
-        return
+      this.loading = true
+      const highlightedFileIsShare =
+        this.isSharedWithMeLocation ||
+        this.isSharedWithOthersLocation ||
+        this.isSharedViaLinkLocation
+
+      if (loadShares) {
+        this.loadShares()
       }
 
-      if (
-        isLocationTrashActive(this.$router, 'files-trash-personal') ||
-        isLocationTrashActive(this.$router, 'files-trash-spaces-project') ||
-        this.highlightedFileIsSpace
-      ) {
-        if (loadShares) {
-          this.loadShares()
-        }
+      if (!highlightedFileIsShare) {
         this.selectedFile = { ...this.highlightedFile }
+        this.loading = false
         return
       }
 
-      this.loading = true
+      // shared resources look different, hence we need to fetch the actual resource here
```

Comment:
I always wondered why we have `highlightedFile` AND `selectedFile`. Turns out, it's needed because share resources look different than "normal" resources. Some information like `mdate` is missing in a built share. Therefore we need to fetch the actual resource in such cases.

Maybe the OCS share API will be extended by the missing properties in the future. Then we could skip this whole fetching-stuff here.